src/main/kotlin/org/rust/ide/refactoring/suggested/RsSuggestedRefactoringSupport.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.refactoring.suggested
+
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.refactoring.suggested.*
+import org.rust.ide.refactoring.isValidRustVariableIdentifier
+import org.rust.lang.core.psi.RsPatBinding
+import org.rust.lang.core.psi.RsPatIdent
+
+class RsSuggestedRefactoringSupport : SuggestedRefactoringSupport {
+    override val availability: SuggestedRefactoringAvailability
+        get() = SuggestedRefactoringAvailability.RenameOnly(this)
+    override val execution: SuggestedRefactoringExecution
+        get() = SuggestedRefactoringExecution.RenameOnly(this)
+    override val stateChanges: SuggestedRefactoringStateChanges
+        get() = SuggestedRefactoringStateChanges.RenameOnly(this)
+    override val ui: SuggestedRefactoringUI
+        get() = SuggestedRefactoringUI.RenameOnly
+
+    override fun importsRange(psiFile: PsiFile): TextRange? = null
+
+    override fun isDeclaration(psiElement: PsiElement): Boolean =
+        psiElement is RsPatBinding && psiElement.parent is RsPatIdent
+
+    override fun isIdentifierPart(c: Char): Boolean = isValidRustVariableIdentifier(c.toString())
+    override fun isIdentifierStart(c: Char): Boolean = isValidRustVariableIdentifier(c.toString())
```

Comment:
This seems a bit weird, I didn't understand how exactly is an identifier defined in the lexer - does it include numeric literals? But this implementation seems to work, i.e. the suggestion is not offered for `5`, but is offered for `a5` or `a`, which seems correct.

Reply:
If I understand Java's std API right (I would not rely on myself!), `Character.isUnicodeIdentifierStart()`/`Character.isUnicodeIdentifierPart()` should works here. If it actually works, let's use it because it just faster then the lexer.

The fun fact is that both options don't work with raw identifiers (because the API sucks). But I think we can live without raw identifiers support in such case